### PR TITLE
Highway layer torch

### PR DESCRIPTION
**Pull request by Shiva-sankaran**
## Description

Adds new layer HighwayLayer implemented in PyTorch, required for implementing TextCNNModel in PyTorch. Refer to #2863 for the Tensorflow to PyTorch model conversion list.

## Type of change

Please check the option that is related to your PR.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
  - In this case, we recommend to discuss your modification on GitHub issues before creating the PR
- [ ] Documentations (modification for documents)

## Checklist

- [x] My code follows [the style guidelines of this project](https://deepchem.readthedocs.io/en/latest/development_guide/coding.html)
  - [x] Run `yapf -i <modified file>` and check no errors (**yapf version must be  0.32.0**)
  - [x] Run `mypy -p deepchem` and check no errors
  - [x] Run `flake8 <modified file> --count` and check no errors
  - [x] Run `python -m doctest <modified file>` and check no errors
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New unit tests pass locally with my changes
- [x] I have checked my code and corrected any misspellings

### CODE REVIEW

File: deepchem/models/torch_models/layers.py [hunk 1]
```diff
@@ -6080,3 +6080,78 @@ def forward(self, dist: torch.Tensor, angle: torch.Tensor,
...
+    """
+
+    def __init__(self,
+                 d_input: int,
+                 activation_fn: Union[Callable, str] = 'relu'):
+        """
+        Initializes the HighwayLayer.
+
+        Parameters
+            ----------
```
Comment (rbharath) [inline deepchem/models/torch_models/layers.py:6121]: Make sure to fix spacing

Comment (Shiva-sankaran) [reply deepchem/models/torch_models/layers.py:6120]: Thanks for the review. I am not exactly sure where spacing needs to be fixed. I have made a commit removing some unnecessary new lines and added spacing between the operators. I hope you meant one of these, if not please let me know where spacing needs to be fixed.

Comment (shreyasvinaya) [reply deepchem/models/torch_models/layers.py:6120]: you will have to fix the spacing of the `----` below the parameters so that it lines up properly

File: deepchem/models/tests/test_layers.py [hunk 2]
```diff
@@ -1438,3 +1438,16 @@ def test_mxmnet_sphericalbasis():
                        [-0.0000e+00, -0.0000e+00, -0.0000e+00, -0.0000e+00]])
 
     assert np.allclose(output, result, atol=1e-04)
+
+
+@pytest.mark.torch
+def test_torch_highway_layer():
+    """Test invoking HighwayLayer."""
+    torch.manual_seed(0)
```
Comment (shreyasvinaya) [inline deepchem/models/tests/test_layers.py:1446]: do we need to fix a seed here?
If the test outputs don't pass the assertions sometimes, you can add a `@flaky` tag above the `@pytest.mark.torch`

File: deepchem/models/torch_models/layers.py (hunk 1 above)
Comment (Shiva-sankaran) [reply deepchem/models/torch_models/layers.py]: Thanks, I completely overlooked that one.

File: deepchem/models/tests/test_layers.py (hunk 2 above)
Comment (Shiva-sankaran) [reply deepchem/models/tests/test_layers.py]: Removed it. Its not needed for the current test.

**Review by rbharath — APPROVED** [review]
LGTM
